File: backend/schemas/gallery.py
"""
公司相册 Schema。
"""
from pydantic import BaseModel, Field


class GalleryItem(BaseModel):
    """相册照片"""
    id: int = Field(..., description="照片 ID")
    category: str = Field(..., description="分类标识: office(办公环境) / team(团队活动) / event(公司大事记)")
    categoryName: str = Field(..., description="分类名称")
    title: str = Field(..., description="照片标题")
    image: str = Field(..., description="图片地址")
    description: str = Field(default="", description="描述")

File: src/api/modules/gallery.js
/**
 * 公司相册 / 办公环境 API。
 */
import { get } from '../request.js'

/**
 * 获取公司相册列表，可按分类筛选。
 * @param {string} [category] - 分类筛选（如 'office', 'team', 'event'）
 * @returns {Promise<Array<{
 *   id: number,
 *   category: string,     // 分类标识
 *   categoryName: string, // 分类名称
 *   title: string,        // 照片标题
 *   image: string,        // 图片地址
 *   description: string   // 描述
 * }>>}
 */
export function getGallery(category) {
  return get('/api/gallery', category ? { category } : undefined)
}
